src/main/java/io/vertx/java/spi/cluster/impl/jgroups/domain/ClusteredCounterImpl.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2011-2013 The original author or authors
+ * ------------------------------------------------------
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Apache License v2.0 which accompanies this distribution.
+ *
+ *     The Eclipse Public License is available at
+ *     http://www.eclipse.org/legal/epl-v10.html
+ *
+ *     The Apache License v2.0 is available at
+ *     http://www.opensource.org/licenses/apache2.0.php
+ *
+ * You may elect to redistribute this code under either of these licenses.
+ */
+
+package io.vertx.java.spi.cluster.impl.jgroups.domain;
+
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import org.jgroups.blocks.atomic.Counter;
+
+public class ClusteredCounterImpl implements io.vertx.core.shareddata.Counter {
+
+  private Vertx vertx;
+  private Counter counter;
+
+  public ClusteredCounterImpl(Vertx vertx, Counter counter) {
+    this.vertx = vertx;
+    this.counter = counter;
+  }
+
+  @Override
+  public void get(Handler<AsyncResult<Long>> handler) {
+    vertx.executeBlocking((future) -> future.complete(counter.get()), handler);
```

Comment:
The Vert.x philosophy is to make everything non blocking unless there is really no alternative.
I suggest we get this merged in as is, but it would be great if they could refactored later to use non blocking JGroups core functions. I think this will be important if this becomes the official cluster manager when Vert.x becomes part of a Red Hat product :)
One thing I'm not sure - does JGroups still use blocking IO?